File: backend/src/main/java/com/strokexai/backend/service/PatientService.java
package com.strokexai.backend.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.strokexai.backend.entity.Patient;
import com.strokexai.backend.entity.User;
import com.strokexai.backend.repository.PatientRepository;
import com.strokexai.backend.repository.UserRepository;

@Service
public class PatientService {

    @Autowired
    private PatientRepository patientRepository;

    @Autowired
    private UserRepository userRepository;

    public Patient createPatient(Patient patient, Long doctorId) {
        User doctor = userRepository.findById(doctorId)
                .orElseThrow(() -> new RuntimeException("Doctor not found"));
        patient.setDoctor(doctor);

        // Generate patient ID
        String patientId = "PT-" + System.currentTimeMillis();
        patient.setPatientId(patientId);

        return patientRepository.save(patient);
    }

    public List<Patient> getPatientsByDoctor(Long doctorId) {
        User doctor = userRepository.findById(doctorId)
                .orElseThrow(() -> new RuntimeException("Doctor not found"));
        return patientRepository.findByDoctor(doctor);
    }

    public List<Patient> getAllPatients() {
        return patientRepository.findAll();
    }

    public Patient getPatientById(Long id) {
        return patientRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Patient not found"));
    }

    public Patient updatePatient(Long id, Patient updatedPatient) {
        Patient existing = patientRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Patient not found"));

        existing.setName(updatedPatient.getName());
        existing.setAge(updatedPatient.getAge());
        existing.setGender(updatedPatient.getGender());
        existing.setPhone(updatedPatient.getPhone());
        existing.setAddress(updatedPatient.getAddress());
        existing.setMedicalHistory(updatedPatient.getMedicalHistory());

        return patientRepository.save(existing);
    }

    public void deletePatient(Long id) {
        patientRepository.deleteById(id);
    }
}

File: frontend/src/services/patientService.js
import api from './api';

export const patientService = {

  createPatient: async (patientData, doctorId) => {
    const response = await api.post(
      `/patients/create/${doctorId}`,
      patientData
    );
    return response.data;
  },

  getMyPatients: async (doctorId) => {
    const response = await api.get(`/patients/doctor/${doctorId}`);
    return response.data;
  },

  getAllPatients: async () => {
    const response = await api.get('/patients/all');
    return response.data;
  },

  getPatientById: async (id) => {
    const response = await api.get(`/patients/${id}`);
    return response.data;
  },

  updatePatient: async (id, patientData) => {
    const response = await api.put(`/patients/update/${id}`, patientData);
    return response.data;
  },

  deletePatient: async (id) => {
    const response = await api.delete(`/patients/delete/${id}`);
    return response.data;
  },
};